tools/azsdk-cli/docs/specs/0-core-telemetry.spec.md

```diff
@@ -0,0 +1,1543 @@
+```kusto
+RawEventsDependencies
+| where name == "ToolExecuted"
+| extend language = tostring(customDimensions.language)
+| where isnotempty(language)
+| summarize EventCount = count() by language
+```
+
+**Question**: What is the geographic distribution of users?
+```kusto
+RawEventsDependencies
+| where name in ("ToolExecuted", "SystemStarted")
+| summarize UserCount = dcount(tostring(customDimensions.devDeviceId)) 
+    by client_CountryOrRegion
+| order by UserCount desc
+```
+
+---
+
+## Open Questions
+
+- [x] **Question 1: Custom vs. Standard SDK Tracking**
+  - **Context**: Issue comments ask how to distinguish custom SDKs from standard SDKs
+  - **Options**: 
+    1. Add `isCustomSdk` boolean to telemetry based on presence of customization files
+    2. Infer from tool calls (e.g., if user calls customization-related tools)
+    3. Don't track this distinction initially
+  - **Proposal**: Option 1 - Add `isCustomSdk` boolean field based on detection of custom code files in SDK directory
+
+- [x] **Question 2: NPS Score Collection**
+  - **Context**: Issue asks about NPS tracking but current telemetry doesn't support it
+  - **Options**:
+    1. Add NPS prompt as system event after first successful SDK generation
+    2. Collect NPS via separate survey tool and correlate by `devDeviceId`
+    3. Defer NPS to future iteration
+  - **Proposal**: Option 3 - Defer to future iteration. Focus on usage metrics first.
+
+- [x] **Question 3: Agent Suggestion Tracking**
+  - **Context**: Issue asks about tracking suggestions presented vs. applied vs. overridden
+  - **Options**:
+    1. Track when Copilot suggests a tool (requires Copilot integration)
+    2. Track only executed tools (current approach)
+    3. Add response-based suggestion tracking (tools suggest next steps in responses)
+  - **Proposal**: Option 2 for initial implementation. Option 3 can be added later if tools return structured JSON with suggestions.
+
+- [x] **Question 4: Release Pipeline Integration**
+  - **Context**: Adoption gap calculation requires package release data
+  - **Options**:
+    1. Integrate with Azure DevOps pipeline telemetry
+    2. Integrate with package manager APIs (PyPI, npm, Maven, NuGet)
+    3. Manual correlation initially
+  - **Proposal**: Option 3 initially, plan for Option 1 integration in Phase 2
+
+- [ ] **Question 5: Telemetry Opt-Out UX**
+  - **Context**: Users must be able to disable telemetry
+  - **Options**:
+    1. Environment variable (`AZSDK_TELEMETRY_ENABLED=false`)
+    2. Configuration file setting
+    3. Interactive prompt on first run
+    4. All of the above
+  - **Proposal**: TBD - Need to align with team preferences
+
+- [ ] **Question 6: Sensitive Data Sanitization**
+  - **Context**: Tool arguments may contain secrets, internal URLs, etc.
+  - **Options**:
+    1. Allowlist approach (only include known-safe fields)
+    2. Blocklist approach (redact known patterns like API keys)
```

Comment:
I think we should do a mix of copying over some of the test-proxy sanitizers as well as enabling CLI options/arguments to be annotated as secret.